di-ipv-cimit-stub/lambdas/stub-management/src/main/java/uk/gov/di/ipv/core/stubmanagement/service/impl/UserServiceImpl.java

```diff
@@ -35,40 +37,77 @@ public UserServiceImpl(ConfigService configService, CimitStubService cimitStubSe
     @Override
     public void addUserCis(String userId, List<UserCisRequest> userCisRequest) {
         List<CimitStubItem> cimitStubItems = cimitStubService.getCimitStubItems(userId);
-        if (!cimitStubItems.isEmpty()) {
-            throw new DataAlreadyExistException(
-                    "User already exists, instead try calling update api.");
-        }
         userCisRequest.forEach(
                 user -> {
-                    cimitStubService.persistCimitStub(
-                            userId,
-                            user.getCode(),
-                            getIssuanceDate(user.getIssuenceDate()),
-                            user.getMitigations());
+                    Optional<CimitStubItem> cimitStubItem =
+                            getUserIdAndCodeFromDatabase(cimitStubItems, user.getCode());
+                    if (cimitStubItem.isEmpty()) {
+                        cimitStubService.persistCimitStub(
+                                userId,
+                                user.getCode(),
+                                getIssuanceDate(user.getIssuanceDate()),
+                                user.getMitigations());
+                    } else {
+                        cimitStubItem
+                                .get()
+                                .setMitigations(
+                                        getUpdatedMitigationsList(
+                                                cimitStubItem.get().getMitigations(),
+                                                user.getMitigations()));
+                        cimitStubItem
+                                .get()
+                                .setIssuanceDate(getIssuanceDate(user.getIssuanceDate()));
+                        cimitStubService.updateCimitStub(cimitStubItem.get());
+                    }
                 });
         LOGGER.info("Inserted User CI data to the Cimit Stub DynamoDB Table.");
     }
 
     @Override
     public void updateUserCis(String userId, List<UserCisRequest> userCisRequest) {
+        if (userCisRequest.stream().allMatch(user -> StringUtils.isEmpty(user.getCode()))) {
+            throw new BadRequestException("User's CI Code cannot be null in all CIs");
+        }
```

Comment:
I'd be inclined to delete all, then insert the supplied items - not sure if the TTL gets reset on an update?

Reply:
I deleted items before updating any items and after checking the request. So, we can be sure to delete all items before any updates. As you know, TTL is only updated upon item creation. I have updated it to TTL. When any items is updated.

Reply:
I tested in my dev environment and followed to the TTL after updated to the item.